doc/inlining.md

```diff
@@ -1,35 +1,61 @@
-## Warmup
+## Simple inlining
 
 As a simple warmup example lets consider inlining a function without osrs.
-Using some fresh variables for all variables of the function we simulate a letrec
-block to evaluate the body inside.
+
 
 ```
 function main()
   var x = 1
-  call y = 'foo(x)
+  call y = 'foo(x) -> l
   print y
 
 function foo(var x)
-  return x
+  var t = nil
+  read t
+  branch t $z $x
+  $z:
+    var z = x + x
+    print z
+    return z
+  $x:
+    return x
 ```
 
+(the `-> l` syntax indicates that `l` is the "return label" of the
```

Comment:
I like it for the implementation because it ensures that you cannot accidentially insert instructions between the call and the label.
But for the formalism I find it not necessary and honestly to me it stands out (ie. why do you even need it, there is a label guaranteed to be right after the call...)